### ISSUE TITLE
Headers: Missing functionality

### ISSUE THREAD

**Issue by BelAn97** [jdi-testing/jdi-dark#128]
Following problems were identified:

        No header assertions methods are implemented. All check have to use restassured header() method. The only method to access headers available in JDI-Dark currently is headers(), which returns headers as list of Header objects. 

    Not possible to pass headers as list or hashmap to service object. 

    Multiple value headers are not supported (Name = Value1, Value2, Value3). Not possible to pass non-String value as Header value in service object.  

    Not possible to pass multiple headers with same name ({Header1, Value1}, {Header1, Value2}) 

    Not possible to perform supplying mapping functions of headers, e.g.: header(“SomeHeader”, Integer::parseInt, lessThan(200))  

    Basically no header checks can be performed by Jdi-dark methods or and most of restassured header shenanigans are not available through service object 

Expected:

    Check if every bullet is actual and cannot be resolved with existed framework API
    Find solution for every identified problem remaining in list
    Discuss proposed solution w/Team if someone from review team agrees, create a dedicated task for implementation and link it to the current issue (that include implementation of the proposed solution, test covering and documentation creation)

**Comment by BelAn97** [comment]
done by Iakov
